### PR TITLE
feat(appeal): fetch resources, providers, policies and appeals with errgroup

### PR DESCRIPTION
**Pull request by idilhaq**

### CODE REVIEW

File: core/appeal/service.go [hunk 1]
```diff
@@ -188,25 +188,54 @@ func (s *Service) Create(ctx context.Context, appeals []*domain.Appeal, opts ...
...
-	}
 
-	pendingAppeals, err := s.getAppealsMap(ctx, &domain.ListAppealsFilter{
-		Statuses:   []string{domain.AppealStatusPending},
-		AccountIDs: accountIDs,
+	eg, egctx := errgroup.WithContext(ctx)
+	var resources map[string]*domain.Resource
+	var providers map[string]map[string]*domain.Provider
+	var policies map[string]map[uint]*domain.Policy
+	var pendingAppeals map[string]map[string]map[string]*domain.Appeal
```
Comment (mabdh) [inline core/appeal/service.go:196]: we could club into this
```suggestion
	var (
	    resources map[string]*domain.Resource
	    providers map[string]map[string]*domain.Provider
	    policies map[string]map[uint]*domain.Policy
	    pendingAppeals map[string]map[string]map[string]*domain.Appeal
	)
```

**Review by rahmatrhd — APPROVED** [review]

File: core/appeal/service.go [hunk 2]
```diff
@@ -188,25 +188,56 @@ func (s *Service) Create(ctx context.Context, appeals []*domain.Appeal, opts ...
...
+	    resources map[string]*domain.Resource
+	    providers map[string]map[string]*domain.Provider
+	    policies map[string]map[uint]*domain.Policy
+	    pendingAppeals map[string]map[string]map[string]*domain.Appeal
+	)
+
+	eg.Go(func() error {
+		resourcesData, err := s.getResourcesMap(egctx, resourceIDs)
+		if err != nil {
+			return err
```
Comment (StewartJingga) [inline core/appeal/service.go:203]:
```suggestion
			return fmt.Errorf("error getting resource map: %w", err)
```

let's wrap the error for easier debugging?

File: core/appeal/service.go [hunk 3]
```diff
@@ -188,25 +188,56 @@ func (s *Service) Create(ctx context.Context, appeals []*domain.Appeal, opts ...
...
+		resources = resourcesData
+		return nil
 	})
-	if err != nil {
-		return fmt.Errorf("listing pending appeals: %w", err)
+
+	eg.Go(func() error {
+		providersData, err := s.getProvidersMap(egctx)
+		if err != nil {
+			return err
```
Comment (StewartJingga) [inline core/appeal/service.go:212]:
```suggestion
			return fmt.Errorf("error getting providers map: %w", err)
```

File: core/appeal/service.go [hunk 4]
```diff
@@ -188,25 +188,56 @@ func (s *Service) Create(ctx context.Context, appeals []*domain.Appeal, opts ...
...
+		}
+		providers = providersData
+		return nil
+	})
+
+	eg.Go(func() error {
+		policiesData, err := s.getPoliciesMap(egctx)
+		if err != nil {
+			return err
```
Comment (StewartJingga) [inline core/appeal/service.go:221]:
```suggestion
			return fmt.Errorf("error getting policies map: %w", err)
```

**Review by StewartJingga — APPROVED** [review]

**Pull request merged**